daily-challenges/2024-09/20/min240920.spec.ts

```diff
@@ -0,0 +1,61 @@
+/*
+# Javascript
+## Đề bài:
+Tìm số lớn nhất trong mảng.
+
+Cho một mảng các số nguyên, bạn cần tìm ra số lớn nhất trong mảng đó. Trong bài tập này, bạn sẽ viết một hàm để thực hiện việc tìm kiếm số lớn nhất.
+
+### Yêu cầu:
+- Viết một hàm JavaScript có tên findLargestNumber để tìm số lớn nhất trong một mảng các số nguyên.
+- Nếu mảng rỗng, trả về thông báo "Mảng rỗng".
+*/
+
+function findLargestNumber(arr) {
+    let max = arr[0]
+    for (let i = 1; i < arr.length; i++) {
+        if (arr[i] > max) {
+            max = arr[i]
+        }
+    }
+    return max
+}
+
+let arr = [1,5,2,6,8,4,9,23,5,8,3,17,7]
+let max = findLargestNumber(arr)
+console.log(max)
+
+/*
+# Playwright
+## Đề bài
+Viết code automation cho test case sau:
+- Đi tới trang: https://material.playwrightvn.com/
+- Click vào: Bài học 5: Xử lý mouse event
+- Click vào ô: "Nhấn hoặc nhấn đúp vào đây!"
+- Kiểm tra: số lần nhấn = 1, loại nhấn: đơn, phím kèm theo: không có
+- Double click vào ô: "Nhấn hoặc nhấn đúp vào đây!"
+- Kiểm tra: số lần nhân = 3, loại nhấn: đúp, phím kèm theo: không có
+- Giữ shift và click vào ô: "Nhấn hoặc nhấn đúp vào đây!"
+- Kiểm tra: số lần nhấn = 4, loại nhấn: đơn, phím kèm theo: Shift
+*/
+
+import { test, expect } from '@playwright/test'
+
+test('min240920', async ({ page }) => {
+    await page.goto('https://material.playwrightvn.com/')
+    await page.locator(`//a[normalize-space(text())='Bài học 5: Xử lý mouse event']`).click()
+
+    await page.click(`//div[@id='clickArea']`)
```

Comment:
Có thể đặt thành 1 biến để dùng lại cho đỡ bị lặp.

```suggestion
    const clickArea = page.click(`//div[@id='clickArea']`);
    await clickArea.click();
```

Reply:
Cho em hỏi chút là tại sao khi đặt biến đã có click rồi thì ở bên dưới sao lại cần phải click thêm 1 lần nữa ạ?

Reply:
@minvp  à, ý là em có thể dùng lại ở bên dưới ngắn hơn. 

VD:
const clickArea = page.click(`//div[@id='clickArea']`);
await clickArea.click();

Khi xuống dưới lại cần click, em chỉ cần dùng biến clickArea.click lại thôi.

---
Chưa kể sau này cái xpath có thể thay đổi, em chỉ cần sửa xpath 1 chỗ (lúc khai báo biến) thôi là được.

Reply:
Em dùng theo suggest thì thấy báo lỗi là **TypeError: clickArea.click is not a function**

Reply:
Push code lên nha, anh ngó cho